test/unit/fact_parser_test.rb

```diff
@@ -79,6 +79,26 @@ class FactParserTest < ActiveSupport::TestCase
     end
   end
 
+  test "#find_physical_interface finds physical intervaces" do
+    parser.stub(:get_interfaces, ['eth0', 'eth0.0', 'br0', 'bond0']) do
+      result = parser.send(:find_physical_interface)
```

Comment:
I think you'll either need to pass the array of interface names (`['eth0'...]`) into find_physical_interface by doing `result = parser.send(:find_physical_interface, ['eth0'....])` or to call `parser.interfaces` like the test above.